### PR TITLE
revert bug introduced by commit eef1a87

### PR DESCRIPTION
**Pull request by LukeXuan**
Hi, caused by the commit https://github.com/syl20bnr/spacemacs/commit/eef1a87e98d8258c2a295566d954bb7b391942fb. A bug is introduced.

In the current develop branch the `spacemacs/set-evil-search-module` is called before the `hybrid-mode` is loaded, when the `hybrid-mode-use-evil-search-module` is unbounded, and thus always choose `'evil-search` as the search module. This pull request fixes this bug by reverting the change on that single file.

### CODE REVIEW

**Comment by bmag** [comment]
I'm using hybrid style I'm not experiencing any bug. If I don't specify `hybrid-mode-use-evil-search-module` I get `isearch` as the search module. If I specify it using `dotspacemacs-editing-style` I get `evil-search`:
```elisp
   dotspacemacs-editing-style '(hybrid :variables
                                       hybrid-mode-default-state 'normal
                                       hybrid-mode-enable-evilified-state t
                                       hybrid-mode-enable-hjkl-bindings nil
                                       hybrid-mode-use-evil-search-module t)
```

**Comment by LukeXuan** [comment]
Hi, what I've got in the variable `evil-search-module` is indeed `isearch`. However, what I got by key-bindings for `/` is `evil-ex-search-forward`, this shouldn't be the case. It should be `isearch-forward` instead.

**Comment by LukeXuan** [comment]
It seems that when the first time invoke `(spacemacs/set-evil-search-module dotspacemacs-editing-style)` in the `init-evil` the `hybrid-mode.el` is not loaded therefore `hybrid-mode-use-evil-search-module` is `void`, which leads to setting `'evil-search` as the search module.

I'm not quite familiar with elisp but I think there should be some problem in these codes.

**Comment by LukeXuan** [comment]
Hi, @syl20bnr @bmag any update? Spacemacs these days is annoying me since I should manually patch this bug every time I update it.

**Comment by bmag** [comment]
From what I see, `evil-search-forward` is the expected command for `isearch` module (when `hybrid-use-evil-search-module` is bound and equals `nil`), and `evil-ex-search-forward` is the expected command for `evil-search` module. There are only certain modes where `/` is bound to `isearch` independently of the `evil-search-module`'s value.

With the current code (without reverting eef1a87), I get:
- if `hybrid-mode-use-evil-search-module` is set to `nil` in `dotspacemacs-editing-style`: `evil-search-module` is `isearch` and `/` is bound to `evil-search-forward`, as expected.
- if `hybrid-mode-use-evil-search-module` is set to `t` in `dotspacemacs-editing-style`: `evil-search-module` is `evil-search` and `/` is bound to `evil-ex-search-forward`, as expected.
- if `hybrid-mode-use-evil-search-module` isn't set in `dotspacemacs-editing-style`: `evil-search-module` is `isearch` and `/` is bound to `evil-ex-search-forward`, NOT as expected.

After applying this PR, I get:
- if `hybrid-mode-use-evil-search-module` is set to `nil` in `dotspacemacs-editing-style`: `evil-search-module` is `isearch` and `/` is bound to `evil-search-forward`, as expected.
- if `hybrid-mode-use-evil-search-module` is set to `t` in `dotspacemacs-editing-style`: `evil-search-module` is `evil-search` and `/` is bound to `evil-search-forward`, NOT as expected. It should be `evil-search` and `evil-ex-search-forward`.
- if `hybrid-mode-use-evil-search-module` isn't set in `dotspacemacs-editing-style`: `evil-search-module` is `isearch` and `/` is bound to `evil-search-forward`, NOT as expected. the same as setting `hybrid-mode-use-evil-search-module` to `t` (`evil-search` and `evil-ex-search-forward`).

Bottom line, this PR doesn't solve the actual bug, so I'm not going to merge it. However, I believe I found the real source of the bug - `spacemacs/set-evil-search-module`. It only sets `evil-search-module`, while it should call `evil-select-search-module`. PR coming shortly.

**Comment by bmag** [comment]
Correction: when `hybrid-mode-use-evil-search-module` isn't set in `dotspacemacs-editing-style`, then the default should be the same as setting it to `nil`.

**Comment by bmag** [comment]
@LukeXuan can you confirm that #8501 fixes the issue?
